state/execution.go

```diff
@@ -241,17 +240,22 @@ func getBeginBlockValidatorInfo(block *types.Block, lastValSet *types.ValidatorS
 	}
 
 	// determine which validators did not sign last block.
-	signVals := make([]abci.SigningValidator, len(lastValSet.Validators))
+	voteInfos := make([]abci.VoteInfo, len(lastValSet.Validators))
 	for i, val := range lastValSet.Validators {
 		var vote *types.Vote
+		var commitRound = -1
 		if i < len(block.LastCommit.Precommits) {
 			vote = block.LastCommit.Precommits[i]
+			if vote != nil {
+				commitRound = vote.Round
+			}
 		}
-		val := abci.SigningValidator{
-			Validator:       types.TM2PB.ValidatorWithoutPubKey(val),
-			SignedLastBlock: vote != nil,
+		voteInfo := abci.VoteInfo{
+			Validator:       types.TM2PB.Validator(val),
+			SignedLastBlock: vote != nil,        // XXX: should we replace with commitRound == -1 ?
```

Comment:
why send an additional bool if `commit_round=-1` tells u the same thing

Reply:
No good reason.

But I realized I was actually wrong. We do have a single commit round. Its true validators can see commits in different rounds subjectively but when a proposer selects a commit to include in the next block, all teh votes must come from the same round. So we should really revert this